### PR TITLE
[Feature]Support modify iceberg table name/properties/comment

### PR DESCRIPTION
**Pull request by Zhangg7723**
Why I'm doing:

> For better Iceberg format support

What I'm doing:

> Provide new feature for iceberg tables, include rename table/modify table properties/modify table comment.
   In addition, table comment can be defined by create/alter table.
   reconstruct schema changes code.

SQL demo supported:
```
CREATE TABLE ice_table (
  `id` bigint(20) DEFAULT NULL,
  `data` varchar(100) DEFAULT NULL,
  `name` varchar(100) DEFAULT NULL
  )
  COMMENT "first comment"
  PROPERTIES (
  'file_format'='parquet'
 )
alter table ice_table rename new_ice_table;
alter table ice_table set ("file_format"="orc")
alter table ice_table comment = "new comment"
```

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 1]
```diff
@@ -1137,6 +1137,11 @@ private void showCreateExternalCatalogTable(TableName tbl, String catalogName) {
             location = table.getTableLocation();
         }
 
+        // Comment
+        if (!Strings.isNullOrEmpty(table.getComment())) {
+            createTableSql.append("\nCOMMENT (\"").append(table.getComment()).append("\");");
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:1142]: remove the `;`

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java [hunk 2]
```diff
@@ -120,6 +120,12 @@ public boolean dropTable(String dbName, String tableName, boolean purge) {
         return dropped;
     }
 
+    @Override
+    public void renameTable(String dbName, String tblName, String newTblName) throws StarRocksConnectorException {
+        delegate.renameTable(dbName, tblName, newTblName);
+        tables.invalidate(new IcebergTableName(dbName, tblName));
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java:126]: partition names cache also need to invalidate.

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java [hunk 3]
```diff
@@ -398,4 +408,83 @@ public static void applySchemaChanges(UpdateSchema updateSchema, List<AlterClaus
...
+    public static void applyTableChanges(Transaction transaction,
+                                         IcebergCatalog icebergCatalog,
+                                         String dbName,
+                                         String tableName,
+                                         List<AlterClause> tableChanges) {
+        if (transaction == null || transaction.table() == null) {
+            throw new StarRocksConnectorException("Transaction or table cannot be null");
+        }
+
+        for (AlterClause clause : tableChanges) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java:422]: Could you use a design pattern to avoid a lot of if else?

Comment (Zhangg7723) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergApiConverter.java:421]: sure

**Review by stephen-shelby — DISMISSED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Pull request merged**